Source/Particles/Collision/BinaryCollision/DSMC/SplitAndScatterFunc.H

```diff
@@ -88,7 +88,7 @@ public:
         );
 
         amrex::Vector<int> num_added_vec(m_num_product_species, 0);
-        for (int i = 0; i < m_num_product_species; i++)
+        for (int i = 0; i < 2; i++)
```

Comment:
@roelof-groenewald Does DSMC support passing only one species in `<collision_name>.species` (for intra-species scattering)? If yes, this should be generalized...

Reply:
I don't believe that has been tried (at least not by me). I guess generally the plasma simulations would you Coulomb collisions for intra-species scattering, but I could imagine a case where neutral-neutral elastic scattering might be included to capture thermalization.
I guess we can deal with that in a follow-up PR.